pandas/tests/indexes/ranges/test_range.py

```diff
@@ -607,3 +607,11 @@ def test_isin_range(self, base):
         result = base.isin(values)
         expected = np.array([True, False])
         tm.assert_numpy_array_equal(result, expected)
+
+    def test_sort_values_key(self):
+        # GH#43666
+        sort_order = {8: 2, 6: 0, 4: 8, 2: 10, 0: 12}
+        values = RangeIndex(0, 10, 2)
+        result = values.sort_values(key=lambda x: x.map(sort_order))
+        expected = Index([4, 8, 6, 0, 2], dtype="int64")
```

Comment:
@usersblock where did you get this `expected` from?  it doesn't match what we'd get if we converted to Series before doing sort_values